fiat-roles/src/main/groovy/com/netflix/spinnaker/fiat/config/RedisConfig.groovy

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.fiat.config
+
+import com.netflix.spinnaker.fiat.redis.JedisPoolSource
+import com.netflix.spinnaker.fiat.redis.JedisSource
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import redis.clients.jedis.Jedis
+import redis.clients.jedis.JedisPool
+import redis.clients.jedis.Protocol
+
+@Configuration
+@ConditionalOnProperty("redis.connection")
+class RedisConfig {
+
+  @Value('${redis.scheduler: default}')
+  String schedulerType;
+
+  @Value('${redis.parallelism: -1}')
+  Integer parallelism;
+
+  @Bean
+  JedisSource jedisSource(JedisPool jedisPool) {
+    return new JedisPoolSource(jedisPool);
+  }
+
+  @Bean
+  @ConfigurationProperties('redis')
+  GenericObjectPoolConfig redisPoolConfig() {
+    new GenericObjectPoolConfig(maxTotal: 100, maxIdle: 100, minIdle: 25)
+  }
+
+  @Bean
+  JedisPool jedisPool(@Value('${redis.connection:redis://localhost:6379}') String connection,
+                      @Value('${redis.timeout:2000}') int timeout,
+                      GenericObjectPoolConfig redisPoolConfig) {
+    return createPool(redisPoolConfig, connection, timeout)
+  }
+
+  private static JedisPool createPool(GenericObjectPoolConfig redisPoolConfig,
+                                      String connection,
+                                      int timeout) {
+    URI redisConnection = URI.create(connection)
+
+    String host = redisConnection.host
+    int port = redisConnection.port == -1 ? Protocol.DEFAULT_PORT : redisConnection.port
+
+    int database = Integer.parseInt((redisConnection.path ?: "/${Protocol.DEFAULT_DATABASE}").split('/', 2)[1])
+
+    String password = redisConnection.userInfo ? redisConnection.userInfo.split(':', 2)[1] : null
+
+    new JedisPool(redisPoolConfig ?: new GenericObjectPoolConfig(), host, port, timeout, password, database, null)
+  }
+
+  @Bean
+  JedisPool jedisPoolPrevious(
```

Comment:
this was from clouddriver to support reading from an old redis to enable a redeploy that changed the key format - probably not needed in here